extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/ThreadLocalPool.java

```diff
@@ -72,7 +72,7 @@ PoolType pool() {
     }
 
     private final void scanForAbandonedConnections() {
-        for (PoolAndThread pair : allConnections) {
+        for (PoolAndThread pair : new ArrayList<>(allConnections)) {
```

Comment:
I think this may still run into ConcurrentModificationException (when copying to the other list). 

A proper solution is to replace the `ArrayList` with a `CopyOnWriteArrayList`

Reply:
(I used a slightly different approach as I actually did it before seeing your suggestion)

Reply:
And it's actually still not 100% right..

Reply:
Nevermind, should be correct as all accesses to this method are exclusive because of the synchronization pattern.